applications/launchpad/gui-react/src/containers/MiningContainer/MiningContainer.integration.test.tsx

```diff
@@ -0,0 +1,55 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { configureStore } from '@reduxjs/toolkit'
+import { ThemeProvider } from 'styled-components'
+
+import MiningContainer from '.'
+
+import { rootReducer } from '../../store'
+import { initialState as miningInitialState } from '../../store/mining/index'
+import themes from '../../styles/themes'
+import { MiningNodesStatus } from '../../store/mining/types'
+
+describe('MiningContainer with Redux', () => {
+  it('should toggle the node status between running and paused', async () => {
+    render(
+      <Provider
+        store={configureStore({
+          reducer: rootReducer,
+          preloadedState: {
+            mining: {
+              ...miningInitialState,
+              tari: {
+                ...miningInitialState.tari,
+                status: MiningNodesStatus.PAUSED,
+              },
+            },
+          },
+        })}
+      >
+        <ThemeProvider theme={themes.light}>
+          <MiningContainer />
+        </ThemeProvider>
+      </Provider>,
+    )
+    let elRunBtn = screen.getByTestId('tari-run-btn')
+    expect(elRunBtn).toBeInTheDocument()
+
+    fireEvent.click(elRunBtn)
+    await waitFor(() => screen.findByTestId('tari-pause-btn'), {
```

Comment:
I think it should work correctly if you just use `screen.queryByTestId` - it does the polling by itself

Reply:
`queryByTestId` is not async. Also, plain `findByTestId` is async, but it doesn't wait long enough, so that's why I had to use `waitFor`

Reply:
ah, that's because you're testing on the implementation with 2 seconds delay
I haven't realized this earlier

so in the future when we hook up the actual backend, you want to leave this test in?
so in CI we will have to setup the docker images, start the backend for this test and adjust the test / waits for that implementation?
I think in that case you'll have to mock the actual backend communication - if that's the case, you should also mock this here to avoid the 2s delays - and you won't have to waitFor 5s

Reply:
ok, consider me convinced :D